code/modules/mapping/ruins.dm

```diff
@@ -11,8 +11,15 @@
 
 		for(var/turf/check in get_affected_turfs(central_turf,1))
 			var/area/new_area = get_area(check)
-			if(!(istype(new_area, allowed_areas)) || check.flags_1 & NO_RUINS_1)
+			if(check.flags_1 & NO_RUINS_1)
 				valid = FALSE
+			else
+				valid = FALSE // set to false before we check
+				for(var/type in allowed_areas)
+					if(istype(new_area, type)) // it's at least one of our types so it's whitelisted
+						valid = TRUE
+						break
+			if(!valid)
 				break

```

Comment:
```suggestion
			valid = FALSE // set to false before we check
			if(check.flags_1 & NO_RUINS_1)
				break
			for(var/type in allowed_areas)
				if(istype(new_area, type)) // it's at least one of our types so it's whitelisted
					valid = TRUE
					break
		if(!valid)
			break
```

Reply:
This logic is a bit funky. Are you sure it's correct?

Reply:
pretty sure havent seen any ruins generate over the station nor has yogstation